scripts/west_commands/sign.py

```diff
@@ -373,18 +374,18 @@ def edt_flash_params(flash):
                     'DT flash device write-block-size {}'.format(align))
 
         # The partitions node, and its subnode, must provide
-        # the size of image-1 or image-0 partition via the regs property.
-        image_key = 'image-1' if 'image-1' in images else 'image-0'
-        if not images[image_key].regs:
+        # the size of slot1_parrtition or slot0_partition partition via the regs property.
+        slot_key = 'slot0_partition' if 'slot1_partition' in slots else 'slot0_partition'
```

Comment:
```suggestion
        slot_key = 'slot0_partition'
```

Reply:
I would leave the logic as it is currently. Aim of this PR is to switch to DTS nodelabels and keep the rest of the behavior intact. The logic here has been working as is for long time and deserves separate PR.

> Leading on from the suggestion, seems like this should be reworked a lot, the signing script wrongly assumes that slot0 is where the image will execute and wrongly assumes that the slot sizes are equal. When building images and mcuboot for direct slot execution, the slot sizes might be different and the address will not be slot0's address

I have two additional changes coming to the script, but not in this PR.
 1) Additional logic that will check what does the chosen zephyr,code-partition points to instead of assuming slot0_partition is the only one, to warn that single image scenario may have a problem; note that we are not given MCUboot configuration, to check for the options;
 2) Logic that will check whether there is scratch-partition and either warn or error out, depending on its existence, when image being signed fills in any of slots, not leaving required empty pages at the end; again we have no access to MCUboot config here.

Direct XIP is harder here, because we have no way to figure out whether MCUboot uses direct XIP; this may require some Kconfig options to Zephyr that will indicate that app is build for Direct XIP, but direct XIP also means that app may be built for specific slot, and this adds to logic.